### PR TITLE
Reorganize unit test mock data setup for AB#16851 

### PR DESCRIPTION
**Pull request by BrianMaki**
# Implements [AB#16851](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/16851)

## Description

Reorganize unit test setup:

- Reduce use of Record type
- Moved 'Expected' creation from Setup function back to unit test itself for clearer definition
- Moved 'Parameters' to be used when calling service or delegate or repository from Setup function back to unit test itself for clearer definition
- Moved data/object creation from Setup function back to unit test for clearer definition as to what is being used in the mock 
- Removed 'Async' from unit test name
- Created helper functions in unit tests to help reduce duplicated code
- Increased code coverage for v1 UserEmailService and UserSmsService
- Also removed imports no longer being used in code.

See latest Sonar coverage for services: https://sonarcloud.io/code?id=bcgov_healthgateway&selected=bcgov_healthgateway%3AApps

## Testing

- [x] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

## UI Changes

None

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/GatewayApi/test/unit/Services.Test/RegistrationServiceTests.cs [hunk 1]
```diff
@@ -482,23 +485,15 @@ private static CreateUserProfileMock SetupCreateUserProfileMock(
...
                 service,
-                jobServiceMock,
-                createUserRequest,
-                DateTime.UtcNow,
-                jwtEmailAddress);
+                jobServiceMock);
         }
 
-        private static CreateUserProfileExceptionMock SetupCreateUserProfileThrowsExceptionMock(
+        private static IRegistrationService SetupCreateUserProfileThrowsExceptionMock(
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/RegistrationServiceTests.cs:491]: Method name no longer corresponds to the returned type.  I suggest something like `SetupRegistrationServiceForCreateUserProfileThrowsException()`.

File: Apps/GatewayApi/test/unit/Services.Test/UserFeedbackServiceTests.cs [hunk 2]
```diff
@@ -105,84 +178,43 @@ private static Mock<ICryptoDelegate> GetCryptoDelegateMock()
...
+                backgroundJobClientMock.Object,
                 MappingService,
-                new Mock<IAuthenticationDelegate>().Object);
-
-            return new(service, expected, createRating);
+                authenticationDelegateMock.Object);
         }
 
-        private static CreateUserFeedbackMock SetupCreateUserFeedbackMock()
+        private static IUserFeedbackService SetupCreateRatingMock(DbResult<Rating> dbResult, SubmitRating createRating)
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/UserFeedbackServiceTests.cs:204]: Method name no longer corresponds to the returned type.  I suggest something like `SetupUserFeedbackServiceForCreateRating()`.

File: Apps/GatewayApi/test/unit/Services.Test/UserFeedbackServiceTests.cs [hunk 3]
```diff
@@ -105,84 +178,43 @@ private static Mock<ICryptoDelegate> GetCryptoDelegateMock()
...
 
-            DbResult<UserFeedback> dbResult = new()
-            {
-                Payload = userFeedback,
-                Status = DbStatusCode.Created,
-            };
+            return GetUserFeedbackService(ratingDelegateMock: ratingDelegateMock);
+        }
 
+        private static IUserFeedbackService SetupCreateUserFeedbackMock(DbResult<UserFeedback> dbResult, UserFeedback createUserFeedback)
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/UserFeedbackServiceTests.cs:216]: Method name no longer corresponds to the returned type.  I suggest something like `SetupUserFeedbackServiceForCreateUserFeedback()`.

File: Apps/GatewayApi/test/unit/Services.Test/UserPreferenceServiceTests.cs [hunk 4]
```diff
@@ -174,25 +158,57 @@ private static GetUserPreferenceMock SetupGetUserPreferenceMock()
...
+
+        private static IUserPreferenceService GetUserPreferenceService(IMock<IUserPreferenceDelegate> userPreferenceDelegateMock)
+        {
+            return new UserPreferenceService(
+                userPreferenceDelegateMock.Object,
+                MappingService,
+                new Mock<ILogger<UserPreferenceService>>().Object);
+        }
+
+        private static IUserPreferenceService SetupCreateUserPreferenceMock(DbResult<UserPreference> dbResult, UserPreference userPreference)
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/UserPreferenceServiceTests.cs:178]: As above, method name no longer corresponds to the returned type.

File: Apps/GatewayApi/test/unit/Services.Test/UserPreferenceServiceTests.cs [hunk 5]
```diff
@@ -174,25 +158,57 @@ private static GetUserPreferenceMock SetupGetUserPreferenceMock()
...
-            return new(service, expected, Hdid);
+            return GetUserPreferenceService(userPreferenceDelegateMock);
         }
 
-        private sealed record UserPreferenceMock(
-            RequestResult<UserPreferenceModel> Expected,
-            UserPreference UserPreference,
-            DbResult<UserPreference> DbResult,
-            UserPreferenceModel UserPreferenceModel);
+        private static IUserPreferenceService SetupUpdateUserPreferenceMock(DbResult<UserPreference> dbResult, UserPreference userPreference)
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/UserPreferenceServiceTests.cs:191]: As above, method name no longer corresponds to the returned type.

File: Apps/GatewayApi/test/unit/Services.Test/UserPreferenceServiceTests.cs [hunk 6]
```diff
@@ -174,25 +158,57 @@ private static GetUserPreferenceMock SetupGetUserPreferenceMock()
...
+                        It.Is<bool>(x => x == true),
+                        It.IsAny<CancellationToken>()))
+                .ReturnsAsync(dbResult);
 
-        private sealed record CreateUserPreferenceMock(IUserPreferenceService Service, RequestResult<UserPreferenceModel> Expected, UserPreferenceModel UserPreferenceModel);
+            return GetUserPreferenceService(userPreferenceDelegateMock);
+        }
 
-        private sealed record UpdateUserPreferenceMock(IUserPreferenceService Service, RequestResult<UserPreferenceModel> Expected, UserPreferenceModel UserPreferenceModel);
+        private static IUserPreferenceService SetupGetUserPreferenceMock(List<UserPreference> preferences)
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/UserPreferenceServiceTests.cs:204]: As above, method name no longer corresponds to the returned type.

File: Apps/GatewayApi/test/unit/Services.Test/UserPreferenceServiceV2Tests.cs [hunk 7]
```diff
@@ -151,60 +189,53 @@ private static UserPreferenceModel GenerateUserPreferenceModel()
                 HdId = Hdid,
                 Preference = "TutorialPopover",
                 Value = "mocked value",
+                CreatedBy = Hdid,
+                UpdatedBy = Hdid,
             };
         }
 
-        private static UserPreferenceMock SetupUserPreferenceMock(DbStatusCode dbStatusCode)
+        private static IUserPreferenceServiceV2 SetupCreateUserPreferenceMock(DbResult<UserPreference> dbResult, UserPreference createUserPreference)
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/UserPreferenceServiceV2Tests.cs:197]: As above, method name no longer corresponds to the returned type.

File: Apps/GatewayApi/test/unit/Services.Test/UserPreferenceServiceV2Tests.cs [hunk 8]
```diff
@@ -151,60 +189,53 @@ private static UserPreferenceModel GenerateUserPreferenceModel()
...
+                        It.Is<UserPreference>(x => x.Preference == createUserPreference.Preference),
                         It.Is<bool>(x => x == true),
                         It.IsAny<CancellationToken>()))
                 .ReturnsAsync(dbResult);
 
-            IUserPreferenceServiceV2 service = GetUserPreferenceService(userPreferenceDelegateMock);
+            return GetUserPreferenceService(userPreferenceDelegateMock);
+        }
+
+        private static IUserPreferenceServiceV2 SetupUpdateUserPreferenceMock(DbResult<UserPreference> dbResult, UserPreference updateUserPreference)
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/UserPreferenceServiceV2Tests.cs:217]: As above, method name no longer corresponds to the returned type.

File: Apps/GatewayApi/test/unit/Services.Test/UserPreferenceServiceV2Tests.cs [hunk 9]
```diff
@@ -151,60 +189,53 @@ private static UserPreferenceModel GenerateUserPreferenceModel()
...
+                        It.Is<UserPreference>(x => x.Preference == updateUserPreference.Preference),
+                        It.Is<bool>(x => x == true),
+                        It.IsAny<CancellationToken>()))
+                .ReturnsAsync(dbResult);
+
+            return GetUserPreferenceService(userPreferenceDelegateMock);
         }
 
-        private static UserPreferencesMock SetupUserPreferencesMock()
+        private static IUserPreferenceServiceV2 SetupGetUserPreferencesMock(List<UserPreference> preferences)
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/UserPreferenceServiceV2Tests.cs:231]: As above, method name no longer corresponds to the returned type.

File: Apps/GatewayApi/test/unit/Services.Test/UserProfileModelServiceTests.cs [hunk 10]
```diff
@@ -288,17 +289,11 @@ private static Mock<IUserProfileDelegate> SetupUserProfileDelegateMock(
             return userProfileDelegateMock;
         }
 
-        private static UserProfileModelMock SetupUserProfileModelMock(
+        private static IUserProfileModelService SetupBuildUserProfileModelMock(
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/UserProfileModelServiceTests.cs:292]: As above, method name no longer corresponds to the returned type.

File: Apps/GatewayApi/test/unit/Services.Test/UserProfileServiceTests.cs [hunk 11]
```diff
@@ -790,75 +1041,28 @@ private static GetUserProfileMock SetupGetUserProfileMock(
...
-                ? Times.Once()
-                : Times.Never();
-
-            GetUserProfileResult expected = new(userProfileModelResult, timesUpdateUserProfile, betaFeatureMappingErrorMessage);
-            return new(service, userProfileDelegateMock, expected, Hdid, jwtAuthTime);
+            return new(service, userProfileDelegateMock);
         }
 
-        private static UpdateAcceptedTermsMock SetupUpdateAcceptedTermsMock(DbStatusCode updatedStatus, bool userProfileExists)
+        private static IUserProfileService SetupUpdateAcceptedTermsMock(DbResult<UserProfile?> updateResult, UserProfile? userProfile)
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/UserProfileServiceTests.cs:1047]: As above, method name no longer corresponds to the returned type.

File: Apps/GatewayApi/test/unit/Services.Test/UserProfileServiceTests.cs [hunk 12]
```diff
@@ -885,72 +1089,22 @@ private static UpdateAcceptedTermsMock SetupUpdateAcceptedTermsMock(DbStatusCode
...
-                        ResultMessage = !userProfileExists ? "Unable to retrieve user profile" : "Unable to update the terms of service: DB Error",
-                    }
-                    : null,
-            };
-
-            return new(service, expected, Hdid, TermsOfServiceGuid);
         }
 
-        private static CloseUserProfileMock SetupCloseUerProfileMock(DbStatusCode updatedStatus, bool userProfileExists, bool profileClosed)
+        private static CloseUserProfileMock SetupCloseUerProfileMock(
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/UserProfileServiceTests.cs:1102]: Existing typo: `SetupCloseUerProfileMock` should be `SetupCloseUserProfileMock`.

File: Apps/GatewayApi/test/unit/Services.Test/UserProfileServiceTests.cs [hunk 13]
```diff
@@ -1134,56 +1218,11 @@ private static RecoverUserProfileMock SetupRecoverUserProfileMock(DbStatusCode u
...
-
-        private static PhoneNumberShouldBeValidMock SetupPhoneNumberShouldBeValidMock(string phoneNumber, bool valid)
-        {
-            IUserProfileService service = GetUserProfileService(configurationRoot: GetIConfiguration());
-            return new(service, valid, phoneNumber);
+            return new(service, emailQueueServiceMock);
         }
 
-        private static ValidateMinimumAgeMock SetupValidateMinimumAgeMock(int age, int minAge, bool patientErrorExists, bool validAge)
+        private static IUserProfileService SetupValidateMinimumAgeMock(int minAge, RequestResult<PatientModel> patientResult)
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/UserProfileServiceTests.cs:1224]: As above, method name no longer corresponds to the returned type.

File: Apps/GatewayApi/test/unit/Services.Test/UserValidationServiceTests.cs [hunk 14]
```diff
@@ -135,13 +135,7 @@ private static Mock<IPatientDetailsService> SetupPatientDetailsServiceMock(Patie
...
         }
 
-        private static PhoneNumberValidMock SetupPhoneNumberValidMock(string phoneNumber)
-        {
-            IUserValidationService service = GetUserValidationService(configurationRoot: GetIConfiguration());
-            return new(service, phoneNumber);
-        }
-
-        private static ValidateEligibilityMock SetupValidateEligibilityMock(
+        private static IUserValidationService SetupValidateEligibilityMock(
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/UserValidationServiceTests.cs:138]: As above, method name no longer corresponds to the returned type.

File: Apps/AccountDataAccess/test/unit/PatientRepositoryTests.cs [hunk 15]
```diff
@@ -302,110 +346,125 @@ private static BlockAccessMock SetupBlockAccessMock(bool changeFeedEnabled, bool
                 new PatientQueryFactory(new Mock<IServiceProvider>().Object),
                 messageSender.Object);
 
-            return new(patientRepository, blockedAccessDelegate, messageSender, blockedAccess, audit, commit, command);
+            return new(patientRepository, blockedAccessDelegate, messageSender);
         }
 
-        private static CanAccessDatasourceMock SetupCanAccessDatasourceMock(DataSource dataSource)
+        private static IPatientRepository SetupCanAccessDatasourceMock(HashSet<DataSource> dataSources, bool useCache)
```
Comment (MikeLyttle) [inline Apps/AccountDataAccess/test/unit/PatientRepositoryTests.cs:352]: Method name no longer corresponds to the returned type.  I suggest something like `SetupPatientRepositoryForCanAccessDataSource()`.

File: Apps/AccountDataAccess/test/unit/PatientRepositoryTests.cs [hunk 16]
```diff
@@ -302,110 +346,125 @@ private static BlockAccessMock SetupBlockAccessMock(bool changeFeedEnabled, bool
...
                 cacheProviderMock.Object,
                 GetIConfigurationRoot(),
                 new PatientQueryFactory(new Mock<IServiceProvider>().Object),
                 new Mock<IMessageSender>().Object);
-
-            return new(patientRepository, !dataSources.Contains(dataSource), dataSource);
         }
 
-        private static GetBlockedAccessMock SetupGetBlockedAccessMock()
+        private static IPatientRepository SetupGetBlockedAccessMock(BlockedAccess blockedAccess)
```
Comment (MikeLyttle) [inline Apps/AccountDataAccess/test/unit/PatientRepositoryTests.cs:399]: As above, method name no longer corresponds to the returned type.

File: Apps/AccountDataAccess/test/unit/PatientRepositoryTests.cs [hunk 17]
```diff
@@ -302,110 +346,125 @@ private static BlockAccessMock SetupBlockAccessMock(bool changeFeedEnabled, bool
...
                 new Mock<ICacheProvider>().Object,
                 GetIConfigurationRoot(),
                 new PatientQueryFactory(new Mock<IServiceProvider>().Object),
                 new Mock<IMessageSender>().Object);
-
-            return new(patientRepository, blockedAccess, Hdid);
         }
 
-        private static GetDataSourcesMock SetupGetDataSourcesMock()
+        private static IPatientRepository SetupGetDataSourcesMock(HashSet<DataSource> dataSources, bool useCache)
```
Comment (MikeLyttle) [inline Apps/AccountDataAccess/test/unit/PatientRepositoryTests.cs:418]: As above, method name no longer corresponds to the returned type.

File: Apps/AccountDataAccess/test/unit/PatientRepositoryTests.cs [hunk 18]
```diff
@@ -302,110 +346,125 @@ private static BlockAccessMock SetupBlockAccessMock(bool changeFeedEnabled, bool
...
                 cacheProviderMock.Object,
                 GetIConfigurationRoot(),
                 new PatientQueryFactory(new Mock<IServiceProvider>().Object),
                 new Mock<IMessageSender>().Object);
-
-            return new(patientRepository, dataSources, Hdid);
         }
 
-        private static QueryMock SetupQueryMock(PatientDetailSource source, bool useHdid, bool useCache)
+        private static IPatientRepository SetupQueryMock(PatientModel patient, PatientIdentity patientIdentity)
```
Comment (MikeLyttle) [inline Apps/AccountDataAccess/test/unit/PatientRepositoryTests.cs:466]: As above, method name no longer corresponds to the returned type.

File: Apps/AccountDataAccess/test/unit/PatientRepositoryTests.cs [hunk 19]
```diff
@@ -464,89 +523,31 @@ private static QueryMock SetupQueryMock(PatientDetailSource source, bool useHdid
...
                 new Mock<IMessageSender>().Object);
-
-            return new(patientRepository, expected, patientDetailsQuery);
         }
 
-        private static QueryThrowsInvalidOperationExceptionMock SetupQueryThrowsInvalidOperationExceptionMock(
-            string? hdid,
-            string? phn,
-            CancellationToken ct)
+        private static IPatientRepository SetupQueryThrowsInvalidOperationExceptionMock()
```
Comment (MikeLyttle) [inline Apps/AccountDataAccess/test/unit/PatientRepositoryTests.cs:536]: As above, method name no longer corresponds to the returned type.

File: Apps/AccountDataAccess/test/unit/Strategy/HdidAllStrategyTests.cs [hunk 20]
```diff
@@ -164,43 +182,11 @@ private static GetPatientMock SetupGetPatientMock(bool useCache)
                 new Mock<ILogger<HdidAllStrategy>>().Object,
                 Mapper);
 
-            return new(hdidAllStrategy, cacheProvider, patient, patientRequest);
+            return new(hdidAllStrategy, cacheProvider);
         }
 
-        private static GetPatientIdentityMock SetupGetPatientIdentityMock()
+        private static HdidAllStrategy SetupGetPatientFromPhsaMock(PatientIdentity patientIdentity)
```
Comment (MikeLyttle) [inline Apps/AccountDataAccess/test/unit/Strategy/HdidAllStrategyTests.cs:188]: As above, method name no longer corresponds to the returned type.

File: Apps/AccountDataAccess/test/unit/Strategy/HdidAllStrategyTests.cs [hunk 21]
```diff
@@ -217,30 +203,17 @@ private static GetPatientIdentityMock SetupGetPatientIdentityMock()
...
                 clientRegistriesDelegate.Object,
                 patientIdentityApi.Object,
                 new Mock<ILogger<HdidAllStrategy>>().Object,
                 Mapper);
-
-            return new(hdidAllStrategy, patient, patientRequest);
         }
 
-        private static GetPatientHandlesExceptionMock SetupGetPatientHandlesExceptionMock()
+        private static HdidAllStrategy SetupGetPatientHandlesPatientIdentityExceptionMock()
```
Comment (MikeLyttle) [inline Apps/AccountDataAccess/test/unit/Strategy/HdidAllStrategyTests.cs:215]: As above, method name no longer corresponds to the returned type.

File: Apps/AccountDataAccess/test/unit/Strategy/HdidPhsaStrategyTests.cs [hunk 22]
```diff
@@ -97,66 +114,49 @@ private static IConfigurationRoot GetConfiguration()
...
-                .ReturnsAsync(patientIdentity);
-
-            HdidPhsaStrategy hdidPhsaStrategy = GetHdidPhsaStrategy(cacheProvider, patientIdentityApi);
-
-            return new(hdidPhsaStrategy, cacheProvider, patient, patientRequest);
+            return new(hdidPhsaStrategy, cacheProvider);
         }
 
-        private static GetPatientHandlesPhsaApiExceptionMock SetupGetPatientHandlesPhsaApiExceptionMock()
+        private static HdidPhsaStrategy SetupGetPatientHandlesPhsaApiExceptionMock()
```
Comment (MikeLyttle) [inline Apps/AccountDataAccess/test/unit/Strategy/HdidPhsaStrategyTests.cs:157]: As above, method name no longer corresponds to the returned type.

File: Apps/Common/test/unit/Utils/HttpContextHelperTests.cs [hunk 23]
```diff
@@ -45,17 +45,24 @@ public class HttpContextHelperTests
...
+            string? actual = HttpContextHelper.GetResourceHdid(httpContextMock.Object, lookupMethod);
 
             // Assert
-            Assert.Equal(mock.Expected, actual);
+            Assert.Equal(expected, actual);
         }
 
         [SuppressMessage("ReSharper", "SwitchStatementHandlesSomeKnownEnumValuesWithDefault", Justification = "Unknown enum value expected and is handled in mock setup.")]
-        private static GetResourceHdidMock SetupGetResourceHdidMock(FhirSubjectLookupMethod lookupMethod)
+        private static Mock<HttpContext> SetupGetResourceHdidMock(FhirSubjectLookupMethod lookupMethod)
```
Comment (MikeLyttle) [inline Apps/Common/test/unit/Utils/HttpContextHelperTests.cs:65]: As above, method name no longer corresponds to the returned type.

File: Apps/GatewayApi/test/unit/Services.Test/ApplicationSettingsServiceTests.cs [hunk 24]
```diff
@@ -40,21 +39,21 @@ public class ApplicationSettingsServiceTests
...
-            DateTime? actual = await mock.ApplicationSettingsService.GetLatestTourChangeDateTimeAsync();
+            DateTime? actual = await service.GetLatestTourChangeDateTimeAsync();
 
             // Assert
-            Assert.Equal(mock.Expected, actual);
+            Assert.Equal(expected, actual);
         }
 
-        private static GetLatestTourChangeDateTimeMock SetupGetLatestTourChangeDateTimeMock(bool useCache)
+        private static IApplicationSettingsService SetupGetLatestTourChangeDateTimeMock(bool useCache, DateTime latestTourDate)
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/ApplicationSettingsServiceTests.cs:55]: As above, method name no longer corresponds to the returned type.

File: Apps/GatewayApi/test/unit/Services.Test/GatewayApiCommunicationServiceTests.cs [hunk 25]
```diff
@@ -113,60 +135,29 @@ private static GetActiveCommunicationMock SetupGetActiveCommunicationMock(
...
                 .ReturnsAsync(requestResult);
 
-            IGatewayApiCommunicationService service = GetGatewayCommunicationService(communicationServiceMock);
-
-            return new(service, expected, communicationType);
+            return GetGatewayApiCommunicationService(communicationServiceMock);
         }
 
-        private static GetActiveCommunicationReturnsErrorMock SetupGetActiveCommunicationReturnsErrorMock()
+        private static IGatewayApiCommunicationService SetupGetActiveCommunicationReturnsErrorMock(string errorMessage)
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/GatewayApiCommunicationServiceTests.cs:145]: As above, method name no longer corresponds to the returned type.

File: Apps/GatewayApi/test/unit/Services.Test/LegalAgreementServiceTests.cs [hunk 26]
```diff
@@ -40,99 +40,90 @@ public class LegalAgreementServiceTests
...
+            ILegalAgreementService service = SetupGetActiveTermsOfServiceMock(legalAgreement);
+
+            // Act
+            Guid? actual = await service.GetActiveLegalAgreementId(LegalAgreementType.TermsOfService);
+
+            // Assert
+            actual.ShouldDeepEqual(expected);
+        }
+
+        private static ILegalAgreementService SetupGetActiveTermsOfServiceMock(LegalAgreement legalAgreement)
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/LegalAgreementServiceTests.cs:117]: As above, method name no longer corresponds to the returned type.

File: Apps/GatewayApi/test/unit/Services.Test/LegalAgreementServiceV2Tests.cs [hunk 27]
```diff
@@ -62,72 +62,66 @@ public async Task ShouldGetActiveTermsOfServiceAsync()
...
+                    Id = LegalAgreementId,
                     LegalText = "Test",
                     LegalAgreementCode = LegalAgreementType.TermsOfService,
                     EffectiveDate = DateTime.UtcNow.Date,
                 }
                 : null;
         }
 
-        private static ActiveTermsOfServiceMock SetupActiveTermsOfServiceMock(bool legalAgreementFound = true, Guid? legalAgreementId = null)
+        private static ILegalAgreementServiceV2 SetupGetActiveTermsOfServiceMock(LegalAgreement? legalAgreement)
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/LegalAgreementServiceV2Tests.cs:113]: As above, method name no longer corresponds to the returned type.

File: Apps/GatewayApi/test/unit/Services.Test/PatientDetailsServiceTests.cs [hunk 28]
```diff
@@ -132,14 +141,20 @@ private static PatientModel GeneratePatientModel(
...
-        private static GetPatientMock SetupGetPatientMock(string identifier, PatientIdentifierType identifierType)
+        private static IPatientDetailsService GetPatientDetailsService(Mock<IPatientRepository> patientRepositoryMock)
+        {
+            return new PatientDetailsService(
+                MappingService,
+                new Mock<ILogger<PatientDetailsService>>().Object,
+                patientRepositoryMock.Object);
+        }
+
+        private static IPatientDetailsService SetupGetPatientMock(PatientModel patientModel, string identifier, PatientIdentifierType identifierType)
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/PatientDetailsServiceTests.cs:152]: As above, method name no longer corresponds to the returned type.

File: Apps/GatewayApi/test/unit/Services.Test/PatientDetailsServiceTests.cs [hunk 29]
```diff
@@ -149,23 +164,10 @@ private static GetPatientMock SetupGetPatientMock(string identifier, PatientIden
...
-                CommonName = patientModel.CommonName,
-                LegalName = patientModel.LegalName,
-            };
-
-            return new(patientDetailsService, expected, identifierType, identifier);
+            return GetPatientDetailsService(patientRepositoryMock);
         }
 
-        private static GetPatientThrowsInvalidDataExceptionMock SetupGetPatientThrowsInvalidDataExceptionMock(
+        private static IPatientDetailsService SetupGetPatientThrowsInvalidDataExceptionMock(
```
Comment (MikeLyttle) [inline Apps/GatewayApi/test/unit/Services.Test/PatientDetailsServiceTests.cs:170]: As above, method name no longer corresponds to the returned type.

File: Apps/GatewayApi/test/unit/Services.Test/RegistrationServiceTests.cs (hunk 1 above)
Comment (BrianMaki) [reply Apps/GatewayApi/test/unit/Services.Test/RegistrationServiceTests.cs:491]: done

File: Apps/GatewayApi/test/unit/Services.Test/UserFeedbackServiceTests.cs (hunk 2 above)
Comment (BrianMaki) [reply Apps/GatewayApi/test/unit/Services.Test/UserFeedbackServiceTests.cs:204]: done

File: Apps/GatewayApi/test/unit/Services.Test/UserFeedbackServiceTests.cs (hunk 3 above)
Comment (BrianMaki) [reply Apps/GatewayApi/test/unit/Services.Test/UserFeedbackServiceTests.cs:216]: done

**Review by MikeLyttle — APPROVED** [review]

**Pull request merged**